src/sidebars/favourites/favourite-controller.js

```diff
@@ -129,25 +131,25 @@
 
                             // Repeat the check as in the mean time a stock for this favourite could have been added.
                             if (this.stocks.map((stock1) => { return stock1.code; }).indexOf(favourite) === -1) {
-                                var data = stock.data[0],
+                                var data = stock && stock.data && stock.data[0],
```

Comment:
in this case it'll only set it to a boolean if either of those is one.
In normal use cases it will either be what we want (the data object) or undefined.
Further down there's  `if(data)...` that cares care of either adding the stock data or creating an error entry